src/harness/testcases/WINNF_FT_S_PPR_testcase.py

```diff
@@ -511,24 +515,34 @@ def generate_PPR_3_default_config(self, filename):
     ppa_record_1 = json.load(
         open(os.path.join('testcases', 'testdata', 'ppa_record_1.json')))
     ppa_record_1['zone']['features'][0]['geometry']['coordinates'] = [[[
-        -97.13, 38.85
-    ], [-97.13, 38.75], [-97.05, 38.75], [-97.05, 38.85], [-97.13, 38.85]]]
+        -97.145, 38.85
+    ], [-97.145, 38.75], [-97.05, 38.75], [-97.05, 38.85], [-97.145, 38.85]]]
     ppa_record_1, pal_records_1 = makePpaAndPalRecordsConsistent(
         ppa_record_1, [pal_record_1], pal_low_frequency, pal_high_frequency,
         'test_user_2')
 
     # Generate FAD records.
     cbsd_records = [device_a, device_c, device_e]
-    grant_record_list = [[grant_a], [grant_c], [grant_e]]
     # Create CBSD reference IDs.
-    cbsd_reference_id1 = generateCbsdReferenceId('test_fcc_id_x',
-                                                 'test_serial_number_x')
-    cbsd_reference_id2 = generateCbsdReferenceId('test_fcc_id_y',
-                                                 'test_serial_number_y')
-    cbsd_reference_ids = [[cbsd_reference_id1, cbsd_reference_id2]]
+    # Note: we prepend 'cbsd/' to ensure that these IDs match with the IDs of
+    # the CBSDs themselves.
+    cbsd_reference_id_a = 'cbsd/' + generateCbsdReferenceId(
+        device_a['fccId'], device_a['cbsdSerialNumber'])
+    cbsd_reference_id_c = 'cbsd/' + generateCbsdReferenceId(
+        device_c['fccId'], device_c['cbsdSerialNumber'])
+    cbsd_reference_id_e = 'cbsd/' + generateCbsdReferenceId(
+        device_e['fccId'], device_e['cbsdSerialNumber'])
+    cbsd_reference_ids = [[
+        cbsd_reference_id_a, cbsd_reference_id_c, cbsd_reference_id_e
+    ]]
```

Comment:
this will not fix the problem :(
the main Id problem is between the SAS UUT injected PPA reference IDs and the ID of CBSD records in SAS UUT dump.
asthe reference model take the injected SAS UUT PPA record as protected intity, and the cbsd reference ids of this PPA don't contain 'cbsd/', butwhen TH pull the dump of SAS UUT, the id of CBSD records contain '/cbsd', so it cann't do the link between them.

I think we need to remove this '/cbsd' either from the cbsd records of the dump or from cbsd['id'] in the if statement of getCbsdsNotPartOfPpaCluster(data.py) :/

Reply:
Ok, this time I think I properly fixed it :) The issue seems to have been in both SAS UUT and the peer SAS TH, and as you correctly pointed out, I only fixed the second one yesterday. This time I fixed the first one by making some changes in `test_harness_objects.py` (please double-check).

I double-checked myself by adding the following code to the end of the `getCbsdsNotPartOfPpaCluster()` function:

```
    logging.info('Examining PPA: %s' % ppa_record)
    logging.info('CBSDs: %s', cbsds)
    logging.info('CBSDs not part of cluster: %s', cbsds_not_part_of_ppa_cluster)
```

and the output showed that the peer SAS PPA has this as its cluster list:
```'cbsd/test_fcc_id_a/d7a9fe1be84243ebdd50c1359cf0630c3d273350', 'cbsd/test_fcc_id_c/33c81df2998eb5d587cea7dede0d35d034e79bcd', 'cbsd/test_fcc_id_e/1a57a6a304f29ccc1a5b814d661a8f8ef235bb9d'```

and although `cbsds` included these three CBSDs, `cbsds_not_part_of_ppa_cluster` was empty.

Similar for the SAS UUT PPA, which has this cluster list: `'cbsd/test_fcc_id_b/b8a0c47f2aed5f2b0222c35cd9b91cff745d0c46', 'cbsd/test_fcc_id_d/1584f2ceda10e1c0f29ad72cccdb339f90aecf40'`

Reply:
Seems like this change does not match with WINNF-TS-0096. In 8.3 CBSD Data Message, the `id` field explicitly specifies to add cbsd/ before $CBSD_REFERENCE_ID but in 8.5.1 PPA Information the CBSDReferenceId  field does not say that.
I agree with @RedaHossino that we may have to modify the `data.py` which might have a bigger impact on other tests or we could just add one function at Line 679 which will remove `cbsd/` from CBSD Record.

Reply:
Bah, you're totally right! I took roughly the approach you suggested, but fixed the cluster list of the peer PPA (after writing out the FAD files) rather than removing "cbsd/" from the CBSD IDs.

Ideally we would fix data.py but doing so would be too disruptive at this point, so we have to live with it for now :/ I'll file a bug to track this issue so that we remember to fix it later.

Reply:
I guess we have to do this for SAS UUT FAD as well because when we pull the FAD using 
`self.sas_uut_fad = getFullActivityDumpSasUut(self._sas, self._sas_admin,
                                                 self.fad_cert, self.fad_key)` this might also not have `cbsd/` in Cluster List of PPA but will have `cbsd/$REFERENCE_ID` in CBSD Record?

Reply:
I don't think so: the reference model will use the objects in `self.protected_entity_records` when computing protection -- it only looks at the FADs to figure out which CBSDs exist. I already added "cbsd/" to the reference IDs in `registerCbsdsAndRequestGrantsWithPpa()`, and it's that object which is added to `self.protected_entity_records`. Please double-check my work but I think it's correct?

Reply:
Sorry, yes you are right!

Reply:
No problem -- I appreciate your very careful review!

Reply:
Kate, I think we have the same issue with the method getAuthorizedGrantsFromDomainProxies(data.py).
as a result during the calculationg of  aggregated intereferences of one PPA the reference model will not exclude the PAL grants of this PPA.
could you check that!?

Reply:
Oooof, this is a rough one. As far as I can tell, we need to use "cbsd/$REFERENCE_ID" when we call the pre-IAP and IAP ref models, but then we need to use "$CBSD_ID" when we actually do the aggregate interference check (based on the implementation of `getAuthorizedGrantsFromDomainProxies()`).

In the ideal world, we would just go fix these functions to be correct... but doing so would cause quite a stir right now.

So the hacky fix that I can think of is to actually have `registerCbsdsAndRequestGrantsWithPpa()` return two versions of the PPA, one for each ID type. Then we manage the values in `self.protected_entity_records` carefully so that the right version is used at each step.

This is a really ugly fix, but I don't see anything better which doesn't involve touching other code. Do you? And do you think this will be enough to solve the problem?

Reply:
yeah, I agree the simplest and most clean scenario is to update the code that impact all the xPR and MCP.
your solution is good if we shouldn't modify the MCP related code.
another idea, 
you don't think that after the CPAS step, in the PPR test code we can delete the 'cbsd/' from the  reference ids of PPAs in protected_entity_records and maybe from other record ids if necessary?

Reply:
If $CBSD_ID == $REFERENCE_ID for all SASes, then your suggestion would work. But since the values *can* be different, I think we have to be a bit more careful. I'll update according to my original suggestion ASAP.